### ISSUE TITLE
Allow asciibinder to run in a subdirectory of a git repo

### ISSUE THREAD

**Issue by indigo423** [redhataccess/ascii_binder#76]
AsciiBinder requires to be his own git repository. In our project the documentation is versioned and released with the source code. Asciibinder should be able to use the git branches from the project source repository while the documentation is just a subdirectory of it.

Using asciibinder in the following directory structure gives an 

```
.
├── .git
├── README.adoc
├── doc
│   └── mydoc.adoc
└── src
```

When you run asciibinder in the `doc` directory does not work and gives an error message, the `doc` directory is not an asciibinder repository.

**Comment by bexelbie** [comment]
Are you willing to store _topic_map.yml, _distro_map.yml, etc. in the root of the repo?

**Comment by indigo423** [comment]
In ideal world I would prefer to have all asciibinder related things in the /doc directory.

**Comment by indigo423** [comment]
@bexelbie I've moved this topic aside for a while and used a separate repo instead to drive the evaluation further. We are getting now closer to migrate to AsciiBinder and tried to get it to work in a repo subdir. I have moved the `_topic_map.yml` and `_distro_map.yml` into the root directory and got also in the error message a hint asciibinder is unable to access the `_template` directory which I have linked to the root directory as well. Unfortunately I was not able to get it to work and get the following error message:

```
Building all distributions.

Building AsciiBinder Doc Project for branch 'master'
Error: Could not generate docs:
Errno::ENOENT: No such file or directory @ dir_s_mkdir - /Users/indigo/Desktop/asciibinder-subdir/_preview/ascii_binder/latest/docs/welcome at
    /Users/indigo/.rvm/rubies/ruby-2.2.5/lib/ruby/gems/2.2.0/gems/ascii_binder-0.1.9/lib/ascii_binder/helpers.rb:647:in `mkdir'
    /Users/indigo/.rvm/rubies/ruby-2.2.5/lib/ruby/gems/2.2.0/gems/ascii_binder-0.1.9/lib/ascii_binder/helpers.rb:647:in `block (4 levels) in generate_docs'
    /Users/indigo/.rvm/rubies/ruby-2.2.5/lib/ruby/gems/2.2.0/gems/ascii_binder-0.1.9/lib/ascii_binder/helpers.rb:643:in `each'
    /Users/indigo/.rvm/rubies/ruby-2.2.5/lib/ruby/gems/2.2.0/gems/ascii_binder-0.1.9/lib/ascii_binder/helpers.rb:643:in `block (3 levels) in generate_docs'
    /Users/indigo/.rvm/rubies/ruby-2.2.5/lib/ruby/gems/2.2.0/gems/ascii_binder-0.1.9/lib/ascii_binder/helpers.rb:639:in `each'
    /Users/indigo/.rvm/rubies/ruby-2.2.5/lib/ruby/gems/2.2.0/gems/ascii_binder-0.1.9/lib/ascii_binder/helpers.rb:639:in `block (2 levels) in generate_docs'
    /Users/indigo/.rvm/rubies/ruby-2.2.5/lib/ruby/gems/2.2.0/gems/ascii_binder-0.1.9/lib/ascii_binder/helpers.rb:582:in `each'
    /Users/indigo/.rvm/rubies/ruby-2.2.5/lib/ruby/gems/2.2.0/gems/ascii_binder-0.1.9/lib/ascii_binder/helpers.rb:582:in `block in generate_docs'
    /Users/indigo/.rvm/rubies/ruby-2.2.5/lib/ruby/gems/2.2.0/gems/ascii_binder-0.1.9/lib/ascii_binder/helpers.rb:546:in `each'
    /Users/indigo/.rvm/rubies/ruby-2.2.5/lib/ruby/gems/2.2.0/gems/ascii_binder-0.1.9/lib/ascii_binder/helpers.rb:546:in `generate_docs'
    /Users/indigo/.rvm/rubies/ruby-2.2.5/lib/ruby/gems/2.2.0/gems/ascii_binder-0.1.9/bin/asciibinder:15:in `call_generate'
    /Users/indigo/.rvm/rubies/ruby-2.2.5/lib/ruby/gems/2.2.0/gems/ascii_binder-0.1.9/bin/asciibinder:291:in `<top (required)>'
    /Users/indigo/.rvm/gems/ruby-2.2.5/bin/asciibinder:23:in `load'
    /Users/indigo/.rvm/gems/ruby-2.2.5/bin/asciibinder:23:in `<main>'
    /Users/indigo/.rvm/gems/ruby-2.2.5/bin/ruby_executable_hooks:15:in `eval'
    /Users/indigo/.rvm/gems/ruby-2.2.5/bin/ruby_executable_hooks:15:in `<main>'.
Try --help for help.
```

I've made a [repository](https://github.com/indigo423/asciibinder-subdir) with the default template and what I've modified.

Any hints about this topic would be helpful.

**Comment by indigo423** [comment]
Resolved with #90
